### ISSUE TITLE
Broken Links (404)

### ISSUE THREAD

**Issue by Drtikso** [jMonkeyEngine/wiki#20]
I just started using JME3 and many of the links in wiki are broken

**Comment by Ali-RS** [comment]
Yes
Below  are some of those broken links gathered thanks to @mitm001 
Feel free to add other broken links to it. I will try to fix them one by one.

jMonkeyEngine 3 Tutorial (1) - Hello SimpleApplication
downloaded the jMonkeyEngine SDK -http://jmonkeyengine.org/wiki/doku.php/
Install the jMonkeyEngine - http://jmonkeyengine.org/wiki/doku.php/

jMonkeyEngine 3 Tutorial (3) - Hello Assets
1.Download the town.zip - http://jmonkeyengine.googlecode.com/files/town.zip
This file has been uploaded to a git hub fork by ali.
https://github.com/Ali-RS/jmonkeyengine/blob/master/jme3testdata/src/main/resources/Scenes/town.zip

jMonkeyEngine 3 Tutorial (4) - Hello Update Loop
Link to user-proposed solutions: http://jmonkeyengine.org/wiki/doku.php/jm3:solutions

jMonkeyEngine 3 Tutorial (5) - Hello Input System
Link to user-proposed solutions: http://jmonkeyengine.org/wiki/doku.php/jm3:solutions

jMonkeyEngine 3 Tutorial (6) - Hello Materials
Materials forum thread - http://jmonkeyengine.org/forum/index.php?topic=14179.0
jME3 Materials documentation (PDF) - http://jmonkeyengine.googlecode.com/files/jME3_materials.pdf
Creating textures in Blender - https://www.blender.org/education-help/tutorials/materials/
Various Material screenshots - http://www.shaders.org/ifw2_textures/whatsin10.htm

**Comment by Ali-RS** [comment]
> MonkeyEngine 3 Tutorial (1) - Hello SimpleApplication
> downloaded the jMonkeyEngine SDK -http://jmonkeyengine.org/wiki/doku.php/
> Install the jMonkeyEngine - http://jmonkeyengine.org/wiki/doku.php/

fixed #19

**Comment by Ali-RS** [comment]
> jMonkeyEngine 3 Tutorial (4) - Hello Update Loop
> Link to user-proposed solutions: http://jmonkeyengine.org/wiki/doku.php/jm3:solutions
> 
> jMonkeyEngine 3 Tutorial (5) - Hello Input System
> Link to user-proposed solutions: http://jmonkeyengine.org/wiki/doku.php/jm3:solutions

fixed #21

**Comment by davidB** [comment]
I just push a bunch of search and replace

**Comment by Ali-RS** [comment]
> jMonkeyEngine 3 Tutorial (6) - Hello Materials
> Materials forum thread - http://jmonkeyengine.org/forum/index.php?topic=14179.0
> jME3 Materials documentation (PDF) - http://jmonkeyengine.googlecode.com/files/jME3_materials.pdf
> Creating textures in Blender - https://www.blender.org/education-help/tutorials/materials/
> Various Material screenshots - http://www.shaders.org/ifw2_textures/whatsin10.htm

fixed #23

**Comment by songproducer** [comment]
Asset packs from here https://docs.jmonkeyengine.org/sdk/sample_code.html:
Broken link: https://docs.jmonkeyengine.org/sdk/sdk/asset_packs.html

**Comment by Ali-RS** [comment]
If you browse wiki from https://jmonkeyengine.github.io/wiki/sdk/sample_code.html#toolbar the link is working for Asset packs

**Comment by songproducer** [comment]
@Ali-RS Thanks.

I found some more from this page:
https://jmonkeyengine.github.io/wiki/sdk.html

[Audio and Effect Track Editing](https://jmonkeyengine.github.io/wiki/sdk/effect_tracks.html)
[Animation and Attachment Bones Editing](https://jmonkeyengine.github.io/wiki/sdk/attachment_bones.html)
[App States and Controls](https://jmonkeyengine.github.io/wiki/sdk/appstates.html)

**Comment by Ali-RS** [comment]
> I found some more from this page:
> https://jmonkeyengine.github.io/wiki/sdk.html
> 
> Audio and Effect Track Editing
> Animation and Attachment Bones Editing
> App States and Controls

Unfortunately i could not find working link for them even in webarchive : 

http://web.archive.org/web/20121015105651/http://jmonkeyengine.org/wiki/doku.php/sdk
http://web.archive.org/web/20150211184141/http://wiki.jmonkeyengine.org/doku.php/sdk

**Comment by mitm001** [comment]
I have went back through the pages up to chapter 9 and these are what's left of broken links as of this posting date. First one is from the SDK itself. 

I will post more if found in a few weeks.

**SDK** Under Help
**Broken Link** - _Browse Help Online_
http://jmonkeyengine.org/wiki/doku.php/jme3

**WIKI Page** https://jmonkeyengine.github.io/wiki/jme3/rotate.html
**Broken Link** 
_More about Quaternions…_
http://gpwiki.org/OpenGL:Tutorials:Using_Quaternions_to_represent_rotation

**Broken Link** under Interpolating Rotations 
_com.jme.math.Quaternion_
http://javadoc.jmonkeyengine.org/com/jme/math/Quaternion.html

**Broken Link** under Interpolating Rotations
 _com.jme.math.TransformQuaternion_
http://javadoc.jmonkeyengine.org/com/jme/math/TransformQuaternion.html

**Broken Link** under Transformation Matrix 
_scale_
https://jmonkeyengine.github.io/wiki/jme3/scale.html

**Broken Link** under Transformation Matrix 
_translate_
https://jmonkeyengine.github.io/wiki/jme3/translate.html

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_asset.html
**Broken Link** 
_Download the town.zip_
http://jmonkeyengine.googlecode.com/files/town.zip

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_material.html
**Broken Link** under Shininess and Bumpiness(Whats shown below is what links looks like)

image::http://jmonkeyengine.googlecode.com/svn/trunk/engine/test-data/Textures/Terrain/Pond/Pond.jpg[Pond.jpg,with="64",height="64",align="right"]

image::http://jmonkeyengine.googlecode.com/svn/trunk/engine/test-data/Textures/Terrain/Pond/Pond_normal.png[Pond_normal.png,with="64",height="64",align="right"]

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_collision.html
**Broken Link** under Sample Code 
_download the town.zip_
http://jmonkeyengine.googlecode.com/svn/trunk/engine/town.zip

**Broken Link** under The Physics-Controlled Scene 
_download it here_
http://jmonkeyengine.googlecode.com/svn/trunk/engine/town.zip

**Comment by Ali-RS** [comment]
> SDK Under Help
> Broken Link - Browse Help Online
> http://jmonkeyengine.org/wiki/doku.php/jme3

Fixed for next SDK releases. PR [#104](https://github.com/jMonkeyEngine/sdk/pull/104)

> WIKI Page https://jmonkeyengine.github.io/wiki/jme3/rotate.html
> Broken Link
> More about Quaternions…
> http://gpwiki.org/OpenGL:Tutorials:Using_Quaternions_to_represent_rotation
> 
> Broken Link under Interpolating Rotations
> com.jme.math.Quaternion
> http://javadoc.jmonkeyengine.org/com/jme/math/Quaternion.html

Fixed.

> Broken Link under Interpolating Rotations
> com.jme.math.TransformQuaternion
> http://javadoc.jmonkeyengine.org/com/jme/math/TransformQuaternion.html
> 
> Broken Link under Transformation Matrix
> scale
> https://jmonkeyengine.github.io/wiki/jme3/scale.html
> 
> Broken Link under Transformation Matrix
> translate
> https://jmonkeyengine.github.io/wiki/jme3/translate.html

Not available

**Comment by Ali-RS** [comment]
Uploaded town.zip two wiki in this PR #27 
I will merge and fix broken links to that if there is no objection.

**Comment by mitm001** [comment]
Below are the most recent errors in the wiki as of this posting.

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/intermediate/best_practices.html
**Under** _Debugging and Test Phase_
**Broken Link** Java Assertions http://docs.oracle.com/javase/1.4.2/docs/guide/lang/assert.html

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Wiki Page** https://jmonkeyengine.github.io/wiki/sdk/debugging_profiling_testing.html
**Under** _Creating Tests_
The instructions for setting up the test need to be edited to instruct the user to:

1. Right-click the root folder in the Projects window and choose Set Configuration > Customize. In the Categories menu select Sources. Under the Test Package Folders heading  choose Add Folder > Create New Folder and then rename it as you choose. Click "Open" then "OK" to finish.

Re-number current instructions to 2 through 6.

**Under** _Running Tests_
**Broken Link** Java Assertions http://docs.oracle.com/javase/1.4.2/docs/guide/lang/assert.html

**Under** _Monitoring and Analyzing_
**Typo** under num 4. "you" should be "your"

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/intermediate/simpleapplication.html
**Under** _SimpleApplication Class_
The tip at the bottom of the chart has a typo.

to hide the buil-in screen / to hide the built-in screen 

**Under** _Default Input Mappings_
Missing "to".

KEY_M Debug key: Prints memory usage stats the out stream.
KEY_M Debug key: Prints memory usage stats to the out stream.

Remove "the"

F5 Hides or shows the statistics the bottom left.
F5 Hides or shows the statistics bottom left.

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Wiki Page** https://jmonkeyengine.github.io/wiki/jme3/intermediate/file_types.html
**Under** _jMonkeyEngine3 File Formats_
Missing suffixes in table,

Cell 1 = .j3o
Cell 2 = .j3m
Cell 3 = .j3md
Cell 4 = .j3f

**Under** _Supported External File Types_
Missing suffixes in table,

Cell 1 = .mesh.xml [Ogre Mesh XML]
Cell 2 = .scene	        [Ogre DotScene]
Cell 3 = .obj		[Wavefront]
Cell 4 = .blend		[Blender]
Cell 8 = .dds		[Direct Draw Surface texture]
Cell 9 = .hdr		[High Dynamic Range texture]
Cell 10 = .tga		[Targa Image File texture]
Cell 11 = .pfm		[Portable Float Map texture]
Cell 12  = .fnt??	[AngelCode font]
Cell 13 = .wav		[Wave music]
Cell 14 = .ogg		[OGG Vorbis music]

**Comment by Ali-RS** [comment]
> Wiki Page https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_asset.html
> Broken Link
> Download the town.zip
> http://jmonkeyengine.googlecode.com/files/town.zip
> 
> Wiki Page https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_material.html
> Broken Link under Shininess and Bumpiness(Whats shown below is what links looks like)
> 
> image::http://jmonkeyengine.googlecode.com/svn/trunk/engine/test-data/Textures/Terrain/Pond/Pond.jpg[Pond.jpg,with="64",height="64",align="right"]
> 
> image::http://jmonkeyengine.googlecode.com/svn/trunk/engine/test-data/Textures/Terrain/Pond/Pond_normal.png[Pond_normal.png,with="64",height="64",align="right"]
> 
> Wiki Page https://jmonkeyengine.github.io/wiki/jme3/beginner/hello_collision.html
> Broken Link under Sample Code
> download the town.zip
> http://jmonkeyengine.googlecode.com/svn/trunk/engine/town.zip
> 
> Broken Link under The Physics-Controlled Scene
> download it here
> http://jmonkeyengine.googlecode.com/svn/trunk/engine/town.zip

Fixed

**Comment by Ali-RS** [comment]
> Wiki Page https://jmonkeyengine.github.io/wiki/jme3/intermediate/best_practices.html
> Under Debugging and Test Phase
> Broken Link Java Assertions http://docs.oracle.com/javase/1.4.2/docs/guide/lang/assert.html

Fixed

**Comment by mitm001** [comment]
Hey Ali,
You seem to be the one who has taken up the torch for updating the wiki. I have not learned how to use github or git (outside of just using it for versioning my own code) due to the steep learning curve required to just use jmonkey. I don't yet have the confidence that I am knowledgeable enough to make any changes myself which is why I have been listing all the problems with the wiki here. 

This thread should only be for broken links I feel so should I post the other stuff like typos and code errors in the wiki someplace else?

**Comment by empirephoenix** [comment]
Do not worry, git is not that problematic. basically on github you just do:
1.) fork jme in your own account button should be right top.
2.) clone your copy adjust whatever bugs you find.
3.) on github, after you pushed, make a pull request to jmonkeyengine.

After that it will appear, and can be commented ect. Also there is absolutly no risk of destroying anything, because you only work in a copy, and only after a review your changes are applied to the main repository.

**Comment by mitm001** [comment]
I have forked it from DavidB link above and have tried hitting the edit button on the wiki page and it loaded the page so I will make an attempt next time instead of bothering Ali because I know he's working on his own code and I know how irritating it is to be interrupted.

I didn't notice the tiny buttons at the top of the wiki page actually did anything until just now. They make loading the page easy.

I will definitely make mistakes because there seems to be undocumented changes to the engine that I am finding out about by accident. Things you cant know about unless you have been following jmonkey for awhile.

**Comment by mitm001** [comment]
Is there an "official" SDK repository? I am finding issues with it also.

Edit: By "official" I mean is this it, https://github.com/jMonkeyEngine/sdk

**Comment by Ali-RS** [comment]
> This thread should only be for broken links I feel so should I post the other stuff like typos and code errors in the wiki someplace else?

Yes, i think better to put them in other issue track.

 

> By "official" I mean is this it, https://github.com/jMonkeyEngine/sdk

Yes

**Comment by Ansury** [comment]
**Necro-Issue** (because it's open and why bother with a new one)

There's a few broken on this page: https://jmonkeyengine.github.io/wiki/jme3/advanced/jme3_shaders.html

http://jmonkeyengine.org/groups/development-discussion-jme3/forum/topic/jmonkeyengine3-material-system-full-explanation

http://code.google.com/p/jmonkeyengine/source/browse/trunk/engine/src/core/com/jme3/shader/UniformBinding.java
(Above works, but it's to Google code so I'm pretty sure that's unintended.)

http://code.google.com/p/jmonkeyengine/source/browse/trunk/engine/src/core/com/jme3/scene/VertexBuffer.java
(Same)

Also, the pdf link at the very bottom is completely broken (the link itself, and the anchor).

**Comment by mitm001** [comment]
You can always post it here. 

Fixed

**Comment by RossComputerGuy** [comment]
Found a broken link, https://jmonkeyengine.github.io/wiki/jme3/advanced/ray_and_sweep_tests.html on https://jmonkeyengine.github.io/wiki/jme3.html#physics-integration

**Comment by mitm001** [comment]
Unfortunately there is no page for it and nothing in archives so I commented out the link.

**Comment by SpaceboyRoss01** [comment]
@mitm001 that link is still there

**Comment by mitm001** [comment]
Refresh your browser.

**Comment by RossComputerGuy** [comment]
it's gone now

**Comment by mitm001** [comment]
All links are now fixed in the new antora site. Start a new issue for any new broken links.
